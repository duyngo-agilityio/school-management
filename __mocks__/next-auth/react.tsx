import { createContext } from "react";

export const useSession = () => ({
  data: {
    user: {
      token: "access-token",
      email: "user@gmail.com",
      id: "1",
      name: "User",
    },
    expires: Date.now().toString(),
  },
  status: "authenticated",
  update: async () => null,
});

export type Session = {
  data: {
    user: {
      token: string;
      email: string;
      id: string;
      name: string;
    };
    expires: string;
  };
  status: "loading" | "authenticated" | "unauthenticated";
  update: (data?: Session) => Promise<Session | null>;
};

const INITIAL_SESSION: Session = {
  data: {
    user: {
      token: "access-token",
      email: "user@gmail.com",
      id: "1",
      name: "User",
    },
    expires: Date.now().toString(),
  },
  status: "authenticated",
  update: async () => null,
};

const SessionContext = createContext(INITIAL_SESSION);

export const SessionProvider = ({ children }: React.PropsWithChildren) => {
  <SessionContext.Provider value={INITIAL_SESSION}>
    {children}
  </SessionContext.Provider>;
};
