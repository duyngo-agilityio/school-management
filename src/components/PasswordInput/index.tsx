'use client';

// Libs
import { ChangeEvent, useState } from 'react';

// Components
import { Button, Input, InputGroup, InputRightElement } from '@chakra-ui/react';

// Icons
import { EyeHiddenIcon, EyeShowIcon } from '@/icons';

interface PasswordInputProps {
  value?: string;
  onChange: (e: ChangeEvent<HTMLInputElement>) => void;
  onBlur?: () => void;
}

const PasswordInput = ({ value, onChange, onBlur }: PasswordInputProps) => {
  const [show, setShow] = useState(false);
  const handleClick = () => setShow(!show);

  return (
    <InputGroup size="md">
      <Input
        data-testid="password-input"
        pr="4.5rem"
        type={show ? 'text' : 'password'}
        value={value}
        onChange={onChange}
        onBlur={onBlur}
      />
      <InputRightElement width="4.5rem">
        <Button variant="icon" onClick={handleClick}>
          {show ? <EyeHiddenIcon /> : <EyeShowIcon />}
        </Button>
      </InputRightElement>
    </InputGroup>
  );
};

export default PasswordInput;
