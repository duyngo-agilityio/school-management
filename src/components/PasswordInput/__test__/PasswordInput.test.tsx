import { fireEvent, render } from '@testing-library/react';
import PasswordInput from '..';

const mockOnChange = jest.fn();

const PasswordInputProps = {
  value: 'Abc@123',
  onChange: mockOnChange,
};

const PasswordInputComponent = () =>
  render(<PasswordInput {...PasswordInputProps} />);

describe('PasswordInput Component', () => {
  it('should match snapshot for PasswordInput', () => {
    const { container } = PasswordInputComponent();

    expect(container).toMatchSnapshot();
  });

  it('toggles the visibility of the password when the button is clicked', () => {
    const { getByRole, container } = PasswordInputComponent();

    const inputElement = container.querySelector('input');
    const buttonElement = getByRole('button');

    expect(inputElement).toHaveAttribute('type', 'password');

    fireEvent.click(buttonElement);
    expect(inputElement).toHaveAttribute('type', 'text');

    fireEvent.click(buttonElement);
    expect(inputElement).toHaveAttribute('type', 'password');
  });
});
