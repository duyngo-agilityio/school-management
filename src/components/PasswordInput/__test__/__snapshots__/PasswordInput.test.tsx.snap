// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordInput Component should match snapshot for PasswordInput 1`] = `
<div>
  <div
    class="chakra-input__group css-bx0blc"
    data-group="true"
  >
    <input
      class="chakra-input css-1n0hj84"
      data-testid="password-input"
      type="password"
      value="Abc@123"
    />
    <div
      class="chakra-input__right-element css-vcnsx7"
    >
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        <svg
          fill="none"
          height="16"
          viewBox="0 0 24 24"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 12C1 12 5 4 12 4C19 4 23 12 23 12"
            stroke="#13296A"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M1 12C1 12 5 20 12 20C19 20 23 12 23 12"
            stroke="#13296A"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <circle
            cx="12"
            cy="12"
            r="3"
            stroke="#13296A"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
